src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/config/WebClientConfiguration.kt

```diff
@@ -12,29 +12,39 @@ import org.springframework.security.oauth2.client.web.reactive.function.client.S
 import org.springframework.web.reactive.function.client.WebClient
 
 @Configuration
-class WebClientConfiguration(@Value("\${api.base.url.nomis}") val nomisApiBaseUri: String, @Value("\${api.base.url.oauth}") val oauthApiBaseUri: String) {
+class WebClientConfiguration(
+  @Value("\${api.base.url.nomis}") val nomisApiBaseUri: String,
+  @Value("\${api.base.url.mapping}") val mappingBaseUri: String,
+  @Value("\${api.base.url.oauth}") val oauthApiBaseUri: String
+) {
 
   @Bean
-  fun nomisApiHealthWebClient(): WebClient {
-    return WebClient.builder()
-      .baseUrl(nomisApiBaseUri)
-      .build()
-  }
+  fun nomisApiHealthWebClient(): WebClient = WebClient.builder().baseUrl(nomisApiBaseUri).build()
+
+  @Bean
+  fun mappingHealthWebClient(): WebClient = WebClient.builder().baseUrl(mappingBaseUri).build()
+
+  @Bean
+  fun oauthApiHealthWebClient(): WebClient = WebClient.builder().baseUrl(oauthApiBaseUri).build()
 
   @Bean
-  fun oauthApiHealthWebClient(): WebClient {
+  fun nomisApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId("system-api")
+
     return WebClient.builder()
-      .baseUrl(oauthApiBaseUri)
+      .baseUrl(nomisApiBaseUri)
+      .apply(oauth2Client.oauth2Configuration())
       .build()
   }
 
   @Bean
-  fun nomisApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+  fun mappingWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
     val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
-    oauth2Client.setDefaultClientRegistrationId("nomis-api")
+    oauth2Client.setDefaultClientRegistrationId("system-api")
```

Comment:
I thought it makes sense for mapping to share the client id wtih nomis api, which means it needs the READ_NOMIS role adding (which Ive done in dev)

Reply:
Actually, our convention is a set of client credentials per system a service connects to: so for instance the migration service uses 3 sets; one for nomis api, one for VSIP and one for mapping service.

The Auth team are aware of that pattern and are happy to create new credentials for each environment so suggest requesting that and changing back to `nomis-api` and add a new client registration